packages/contracts/src/test/Integration.t.sol

```diff
@@ -57,14 +90,16 @@ contract Integration is Test {
         // TODO
         //game.createGame(2, game.allowAnyPlayerAndDeck);
         game.createGame(2);
-        inventory.getDeck(player1, 0);
+        inventory.getDeck(player2, 0); // player1 has card id of 49-72 inclusive
+        inventory.getDeck(player2, 0); // player2 has card id of 73-96 inclusive
+        // ids are not starting at 1 because the deploy script currently airdrops some addresses, might change
         vm.startPrank(player1);
         game.commitSalt(COMMITTED_SALT);
-        game.joinGame(gameID, 0, bytes(""), HAND_ROOT, DECK_ROOT, PROOF);
+        game.joinGame(gameID, 0, bytes(""), HAND_ROOT, DECK_ROOT, generateProof());
```

Comment:
Currently only test the first proof verification. When we set the `checkProof` variable to be true, this verification passes. Do you know if there is a foundry cheatcode that can change/set the value of a variable in the smart contract? (So that we can set the value of `checkProof` in the test)

Something like [this](https://github.com/defi-wonderland/smock) but for Foundry.

Reply:
There is load for setting a storage slot: https://book.getfoundry.sh/cheatcodes/load but you have to compute the position of the storage slot yourself.

Another option is add a function in Game.sol that checks the caller is 0x0 then toggles the checkProof value, then impersonate 0x0 with a cheatcode.

Yet another option is have checkProof be a parameter in the Game.sol constructor, and write a function in the deploy script that controls the value of that parameter.

I feel like the second option is convenient for now, not too much code and resistant to refactorings. Might be want to get rid of it in the future if we run into contract size issues (but maybe that's super unavoidable and we'll need to factor out Game.sol in multiple contracts anyway).

Reply:
Yeah the second option sounds good, let's go with that